packages/lisk-p2p/src/p2p.ts

```diff
@@ -111,7 +112,7 @@ export class P2P extends EventEmitter {
 	private readonly _handleOutboundSocketError: (error: Error) => void;
 	private readonly _handleInboundSocketError: (error: Error) => void;
 
-	public constructor(config: P2PConfig) {
+	public constructor(config: P2PConfig, peerSelectionFunction?: P2PPeerSelectionFunctions) {
```

Comment:
name should be `peerSelectionFunctions`?
Also, should we put it in the config then we don't need `peerSelectionFunctions` and we can put one by one

Reply:
I would also prefer them to be in the `P2PConfig` individually as top-level optional fields.